### PR TITLE
[type] Improve GlobalStoreStmt precision on custom floats

### PR DESCRIPTION
**Pull request by yuanming-hu**
Related issue = #1905

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----

### CODE REVIEW

File: taichi/runtime/llvm/runtime.cpp [hunk 1]
```diff
@@ -1585,6 +1585,29 @@ DEFINE_SET_PARTIAL_BITS(8);
...
+    return f + 0.5;
+  else
+    return f - 0.5;
+  */
+
+  // Branch-free implementation: copy the sign bit of "f" to "0.5"
+  i32 delta_bits =
+      (taichi_union_cast<i32>(f) & 0x80000000) | taichi_union_cast<i32>(0.5f);
+  f32 delta = taichi_union_cast<f32>(delta_bits);
+  return f + delta;
```
Comment (Hanke98) [inline taichi/runtime/llvm/runtime.cpp:1601]: It's cool to eliminate branch in this way! Now, I am thinking about is it possible and beneficial to implement this part in `Codegen`? Maybe we can discuss this later.

**Review by Hanke98 — APPROVED** [review]
Cool! Looks great. I only have one off topic comment here and please free to merge this pr. Thanks!

**Review by TH3CHARLie — APPROVED** [review]
Cool branch elimination!

File: taichi/runtime/llvm/runtime.cpp (hunk 1 above)
Comment (yuanming-hu) [reply taichi/runtime/llvm/runtime.cpp:1601]: That's a good point. I'm just being lazy here. There are two benefits of the current implementation:
- Easy to write
- Easy for other backend authors to learn what's happening here

The cons are
- Need to instantiate for `f32`/`f64` separately
- Slower compilation

We can keep the discussion to our future meetings if you are interested in this :-)

**Pull request merged**